app/controllers/AccountManagement.scala

```diff
@@ -166,56 +166,59 @@ object ManageWeekly extends ContextLogging {
-                case Left(error) =>
-                  info(s"couldn't get new rate/currency for renewal: $error")
-                  Ok(views.html.account.details(None, promoCode))
-                case Right((existingCurrency, weeklyPlanInfoList)) =>
-                  Ok(weeklySubscription.asRenewable.map { renewableSub =>
-                    info(s"sub is renewable - showing weeklyRenew page")
-                    views.html.account.weeklyRenew(renewableSub, contact, zuoraContact.email, billToCountry, weeklyPlanInfoList, existingCurrency, promoCode)
-                  } getOrElse {
-                    info(s"sub is not renewable - showing weeklyDetails page")
-                    views.html.account.weeklyDetails(weeklySubscription, billingSchedule, contact)
-                  })
-              }
-            }.getOrElse {
-              info(s"no valid bill to country for account")
+      EitherT(zuoraRest.getAccount(weeklySubscription.accountId)).map { account =>
+        account.billToContact.country.map { billToCountry =>
+          val catalog = tpBackend.catalogService.unsafeCatalog
+          val weeklyPlans = weeklySubscription.planToManage.product match {
+            case Product.WeeklyZoneA => catalog.weekly.zoneA.plans
+            case Product.WeeklyZoneB => catalog.weekly.zoneB.plans
+            case Product.WeeklyZoneC => catalog.weekly.zoneC.plans
+          }
+
+          val renewalPlans = weeklyPlans.filter(_.availableForRenewal)
+
+          val currency = account.currency.toRight(s"could not deserialise currency for account ${account.id}")
+          val weeklyPlanInfo = currency.right.flatMap { existingCurrency =>
+            sequence(renewalPlans.map { plan =>
+              val price = plan.charges.price.getPrice(existingCurrency).toRight(s"could not find price in $existingCurrency for plan ${plan.id} ${plan.name}").right
+              price.map(price => WeeklyPlanInfo(plan.id, plan.charges.prettyPricing(price.currency)))
+            }).right.map(r => (existingCurrency, r))
+          }
+          weeklyPlanInfo match {
+            case Left(error) =>
+              info(s"couldn't get new rate/currency for renewal: $error")
               Ok(views.html.account.details(None, promoCode))
-            }
+            case Right((existingCurrency, weeklyPlanInfoList)) =>
+              Ok(weeklySubscription.asRenewable.map { renewableSub =>
+                info(s"sub is renewable - showing weeklyRenew page")
+                views.html.account.weeklyRenew(renewableSub, account.soldToContact, account.billToContact.email, billToCountry, weeklyPlanInfoList, existingCurrency, promoCode)
+              } getOrElse {
+                info(s"sub is not renewable - showing weeklyDetails page")
+                views.html.account.weeklyDetails(weeklySubscription, billingSchedule, account.soldToContact)
+              })
           }
+        }.getOrElse {
+          info(s"no valid bill to country for account")
```

Comment:
Yes agreed, there are a lot of things we could do here to improve error handling. My main thought was that if this is logged at error level (rather than info), it'll at least throw to Sentry, which gives us a better chance of spotting the problem in prod.